submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,244 @@
+traffic with in a cluster and will be present in every node.
+
+![UpdatedTopology](./images/submariner-ovn-ic.png)
+
+### SubmarinerRouteAgentPod
+
+The Submariner Route-agent pod running on the active gateway node will be responsible for creating the SubmarinerGWRoute CRs. It will be used
+only for OVN CNI. For every RemoteEndpointCreated event a SubmarinerGWRoute CR will be created. The nextHop will be the interface IP through
+which we can reach the cable driver. In the case of OVN it will be the IP of ovn-k8s-mp0 interface.
+
+The SubmarinerNonGWRoute CRD will also be created by Submariner Route-agent. It will be created per endpoint and will have remoterCIDRS from
+the endpoint. The nextHop will be the transit switch IP of the G/W node. If the transit switch IP is missing this CRD will not be created,
+which means it is a non-IC setup.
+
+The RouteAgent  will have these controllers added to it and the one running in gateway node responds to the CRUD operations of Submariner
+endpoints.
+
+#### SubmarinerGWRoute CR Controller
+
+This controller will be responsible for programming the OVN cluster router, it will react only the active g/w node. When a SubmarinerGWRoute
+CR is created or modified the controller shall create or update a routing policy in OVN cluster router with a priority of 20000, and it should
+redirect any traffic destined to remote CIDR to the ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : 0459f009-3603-47ac-8ee7-9d958540ed31
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.1.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+It also programs a route in the ovn-cluster-router, to route the traffic coming from other zones destined to remote cluster IP range via the
+ovn-k8s-mp0 interface IP.
+
+```bash
+_uuid               : d55185d8-3732-45c1-ae90-4a7f8cd191f7
+bfd                 : []
+external_ids        : {}
+ip_prefix           : "10.132.0.0/16"
+nexthop             : "10.1.1.2"
+options             : {}
+output_port         : []
+policy              : []
+route_table         : ""
+```
+
+#### SubmarinerNonGWRoute Controller
+
+This controller will run in every route agent pod. This controller connects to the OVN DB. When a SubmarinerNonGWRoute CR is created
+in non-g/w node,  it updates the ovn-cluster-route with a router policy using a priority of 20000 to send the traffic to
+the remote cluster via next hop mentioned, which is the transit switch IP to the g/w node. Before adding the route it checks if
+a route exists, if so it skips adding the route again. This is required to prevent duplicate update since there can be more than
+one node in each zone and hence more than one RouteAgent.
+
+```bash
+_uuid               : 22db3005-64c5-4e32-aeb0-642423c30742
+action              : reroute
+external_ids        : {}
+match               : "ip4.dst==10.132.0.0/16"
```

Comment:
```suggestion
match               : "ip4.dst==10.132.0.0/14"
```

The diagram uses 10.132.0.0/14 CIDR, lets keep it consistent.

Reply:
Done.